From: Danilo Egea Gondolfo <danilogondolfo@gmail.com>
Date: Tue, 23 Jan 2024 13:41:27 +0000
Subject: tests/sriov: test if the generated netplan-rebind service is correct

This test is currently failing due to interface duplicates being added
to the list.

Origin: https://github.com/canonical/netplan/pull/437
---
 tests/test_sriov.py | 36 ++++++++++++++++++++++++++++++++++++
 1 file changed, 36 insertions(+)

--- a/tests/test_sriov.py
+++ b/tests/test_sriov.py
@@ -879,6 +879,42 @@
 ExecStart=/usr/sbin/netplan rebind enblue engreen
 '''})
 
+    def test_rebind_service_generation(self):
+        self.generate('''network:
+  version: 2
+  ethernets:
+    engreen:
+      embedded-switch-mode: switchdev
+      delay-virtual-functions-rebind: true
+    enblue:
+      match: {driver: fake_driver}
+      set-name: enblue
+      embedded-switch-mode: legacy
+      delay-virtual-functions-rebind: true
+      virtual-function-count: 4
+    sriov_blue_vf0:
+      link: enblue
+    sriov_blue_vf1:
+      link: enblue
+    sriov_blue_vf1:
+      link: enblue
+    sriov_green_vf0:
+      link: engreen
+    sriov_green_vf1:
+      link: engreen
+    sriov_green_vf2:
+      link: engreen''')
+        self.assert_sriov({'rebind.service': '''[Unit]
+Description=(Re-)bind SR-IOV Virtual Functions to their driver
+After=network.target
+After=sys-subsystem-net-devices-enblue.device
+After=sys-subsystem-net-devices-engreen.device
+
+[Service]
+Type=oneshot
+ExecStart=/usr/sbin/netplan rebind enblue engreen
+'''})
+
     def test_rebind_not_delayed(self):
         self.generate('''network:
   version: 2
